### PR TITLE
fix: clean server responses

### PR DESCRIPTION
**Pull request by Nick42791**
This is very experimental.
It solves the problem of having that switch case in the response.ts game api.

May need to be cleaned up a little but it's a start.

### CODE REVIEW

File: client/src/store/mutations/player.ts [hunk 1]
```diff
@@ -11,6 +11,10 @@ function SET_PLAYER_ROLE(state: State, payload: Role) {
   state.role = payload;
 }
 
+function UPDATE_PLAYER_ROLE(state: State, payload: {data: Role, role: Role}) {
```
Comment (alee) [inline client/src/store/mutations/player.ts:14]: hey @Nick42791 what's the purpose of this new mutation?

File: client/src/api/game/response.ts [hunk 2]
```diff
@@ -13,14 +13,18 @@ import {
...
-
+import Mutations from "@port-of-mars/client/store/mutations";
 type Schemify<T> = T & Schema;
 
 function deschemify<T>(s: Schemify<T>): T {
   return s.toJSON() as T;
 }
 
-const responseMap = {
+type serverResponse = {
```
Comment (alee) [inline client/src/api/game/response.ts:23]: LGTM!

File: client/src/store/mutations/player.ts (hunk 1 above)
Comment (cpritcha) [reply client/src/store/mutations/player.ts:14]: Was it get around the type system constraint that you needed a role mutation? You can get around this with something like

```typescript
type ServerResponse = {
  [key in Exclude<keyof PlayerData, 'role'>]:  keyof typeof Mutations
}
```

https://www.typescriptlang.org/docs/handbook/utility-types.html#excludetu

Comment (Nick42791) [reply client/src/store/mutations/player.ts:14]: Calvin was exactly right. I'll fix it really quick. Thanks!

Comment (Nick42791) [reply client/src/store/mutations/player.ts:14]: I'll put up a commit with the logic that removes role. and we can try that out too.

**Pull request merged**